src/components/20-molecules/datepicker/e2e.js

```diff
@@ -36,5 +36,55 @@ test.describe('datepicker', () => {
         '29.02.2020 (of 1 submittable elements)'
       );
     });
+
+    test('should just add years of a given range', async ({ page }) => {
+      await page.goto(fixtureURL('components-datepicker--datepicker'));
+
+      const datepicker = await page.locator('axa-datepicker');
+      await datepicker.evaluate(node => {
+        node.allowedyears = ['1999-2000'];
+      });
+
+      const datepickerYearDropdown = await page.locator(
+        'axa-datepicker .js-datepicker__dropdown-year'
+      );
+      const itemsString = await datepickerYearDropdown.getAttribute('items');
+
+      const itemsArray = JSON.parse(itemsString);
+
+      expect(itemsArray.length).toBe(2);
+      expect(itemsString).toContain('"value":"1999"');
+      expect(itemsString).toContain('"value":"2000"');
+      expect(itemsString).not.toContain('"value":"2020"');
+    });
+
+    test('should fire the right events', async ({ page }) => {
+      await page.goto(
+        fixtureURL('examples-datepicker-pure-html--with-onchange-handler')
+      );
+
+      const datepickerForm = await page.locator(
+        `axa-datepicker[data-test-id="datepicker-onchange"]`
```

Comment:
AFAIK data-test-id values are never reused, so you can simplify this selector by dropping the `axa-datepicker` part.